numpy/lib/shape_base.py

```diff
@@ -757,12 +758,64 @@ def array_split(ary, indices_or_sections, axis=0):
     >>> x = np.arange(8.0)
     >>> np.array_split(x, 3)
     [array([0.,  1.,  2.]), array([3.,  4.,  5.]), array([6.,  7.])]
-
-    >>> x = np.arange(9)
-    >>> np.array_split(x, 4)
-    [array([0, 1, 2]), array([3, 4]), array([5, 6]), array([7, 8])]
-
+    >>> x = np.arange(7.0)
+    >>> np.array_split(x, 3)
+    [array([0.,  1.,  2.]), array([3.,  4.]), array([5.,  6.])]
+    >>> x = np.reshape(np.arange(16), (4, 4))
+    >>> np.array_split(x, [3, 3], (0, 1))
```

Comment:
Do you think we should allow 
```suggestion
    >>> np.array_split(x, 3, (0, 1))
```
to mean the same thing?

Reply:
I like the idea and have added the functionality. Based on your proposed ```test_zero_dimensional_split```, I assumed that ```ret = split(matrix, 3, axis=())``` should operate the same as ```ret = split(matrix, [], axis=())``` and therefore do nothing.

Reply:
Yes, "split into three parts on no axes" seems like a legal instruction to do nothing, and an error would just be annoying.